src/dune_pkg/local_package.mli

```diff
@@ -10,6 +10,13 @@ type pin =
 
 type pins = pin Package_name.Map.t
 
+type command_source =
```

Comment:
Reading this code, I'm confused about the difference between the "origin : [`Dune | `Opam]" field of `pin` and this type. I think it would be helpful to add a comment explaining why both are necessary.

Reply:
I added a comment, tell me if that is clearer or if I need to alter the wording :smile:

Reply:
Ok so as far as I can tell the `origin` field of `pin` is what type of pin it is, as dune understands packages pinned in the opam style, but also defines a different type of pinning that is specific to dune. And `command_source` is orthogonal to this. An opam pin can be missing an opam file and use dune commands, and a dune pin can have an opam file and use the commands from that file.

A minor nitpick: I think we can name the cases to be a bit more descriptive. How about `Assume_defaults` and `Opam_file`?

Reply:
I have a slightly different understanding on this for the `origin`: it defines the context on which it is pinned. All the packages I tried (opam file and no opam file), where using the `'Dune` variant because of the `Pin_stanza.decode` function ([source](https://github.com/ocaml/dune/blob/c19917fbea3c61035ec2f83b7780de89409bc08c/src/dune_pkg/pin_stanza.ml#L164)). The `'Opam` variant seems to happen for the deps that are local packages.

I don't see any issue with modifying the name if that makes it clearer :smile: